### PR TITLE
Docs for useDialog and useOverlayTrigger

### PR DESCRIPTION
**Pull request by devongovett**
Depends on #398, #408.

### CODE REVIEW

File: packages/@react-aria/dialog/docs/useDialog.mdx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    }
+  </>;
+}
+
+// Application must be wrapped in an OverlayContainer so that it can be
+// hidden from screen readers when a modal opens.
+<OverlayProvider>
+  <Example />
+</OverlayProvider>
+```
```
Comment (LFDanLu) [inline packages/@react-aria/dialog/docs/useDialog.mdx:211]: There is a lot of stuff happening in this example that you cover in the paragraph above but I wonder if it would be better to only expose the `useDialog` parts (though this would probably take some additional work to be able to hide the rest of the code or to rework the example entirely) and pare down the explanation accordingly (or reword it to point out what useDialog is adding here). To compensate for the rest of the explanation, maybe it would be better to link to the `useModal`/`useOverlay`/etc future docs instead. It would also help consolidate all the info about these other hooks into one place so you don't have to re-explain what each hook does (unless those hooks won't each have a dedicated docs page?).

On the other hand this does give a very comprehensive explanation of how to build a dialog from scratch all in one place, something that would be lost if split up, hrm. Lemme know what you think

File: packages/@react-aria/overlays/docs/useOverlayTrigger.mdx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+## Example
+
+This example shows how to build a typical popover overlay that is positioned relative to
+a trigger button. The content of the popover is a dialog, built 
+with <TypeLink links={dialogDocs.links} type={dialogDocs.exports.useDialog} />.
+
+The popover can be closed by clicking or interacting outside the popover, or by pressing the Escape key.
+This is handled by <TypeLink links={docs.links} type={docs.exports.useOverlay} />.
+Focus is restored back to the button that triggered the popover it when it closes by a
```
Comment (LFDanLu) [inline packages/@react-aria/overlays/docs/useOverlayTrigger.mdx:81]: Maybe "When the popover is closed, focus is restored back to its trigger button by a FocusScope."

File: packages/@react-aria/overlays/docs/useOverlayTrigger.mdx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+&lt;<TypeLink links={focusDocs.links} type={focusDocs.exports.FocusScope} />&gt;.
+
+Content outside the popover is hidden from screen readers 
+by <TypeLink links={docs.links} type={docs.exports.useModal} />.
+This improves the experience for screen reader users by ensuring that they don't
+navigate out of context. This is especially important when the popover is rendered
+into a portal at the end of the document, and the content just before it will be
+unrelated to the original trigger.
```
Comment (LFDanLu) [inline packages/@react-aria/overlays/docs/useOverlayTrigger.mdx:89]: "will be" -> "is"

File: packages/@react-aria/dialog/docs/useDialog.mdx (hunk 1 above)
Comment (LFDanLu) [reply packages/@react-aria/dialog/docs/useDialog.mdx:211]: Same for useOverlayTrigger

Comment (devongovett) [reply packages/@react-aria/dialog/docs/useDialog.mdx:211]: Yeah it is quite complex... Some of the hooks wouldn't be very useful to document on their own though, i.e. the example kinda needs to be a combination of these things. We could create some higher level hooks to combine some of them into a single hook I suppose, but then you lose the option to only include some functionality depending on your use case (or we end up needing to document those parts separately somehow). 🤔

Comment (dannify) [reply packages/@react-aria/dialog/docs/useDialog.mdx:211]: I feel like even if we had higher level hooks that combined these items, we would still need to document them anyway. I'd rather be a little over the top than not have a real life example. Plus using aria by itself already means you are a bit of an advanced user.
